### PR TITLE
Bugfix/198 use v2 in module path to fit major version

### LINKED ISSUE #198: Use v2 in module path to fit major version

### PR DESCRIPTION
**Pull request by meiserloh**
Resolves #198

### CODE REVIEW

File: controllers/doguInstallManager.go [hunk 1]
```diff
@@ -69,10 +69,10 @@ func NewDoguInstallManager(client client.Client, mgrSet *util.ManagerSet, eventR
 
 // Install installs a given Dogu Resource. This includes fetching the dogu.json and the container image. With the
 // information Install creates a Deployment and a Service
-func (m *doguInstallManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) (err error) {
+func (m *doguInstallManager) Install(ctx context.Context, doguResource *k8sv2.Dogu) (err error) {
```
Comment (cesmarvin) [inline controllers/doguInstallManager.go:72]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Install' has too many statements (59 > 40)

File: docs/operations/installing_operator_into_cluster_en.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ The installation from GitHub requires the installation YAML which contains all r
 
 ```
 GITHUB_VERSION=0.0.6
-kubectl apply -f https://github.com/cloudogu/k8s-dogu-operator/releases/download/v${GITHUB_VERSION}/k8s-dogu-operator_${GITHUB_VERSION}.yaml
+kubectl apply -f https://github.com/cloudogu/k8s-dogu-operator/v2/releases/download/v${GITHUB_VERSION}/k8s-dogu-operator_${GITHUB_VERSION}.yaml
```
Comment (jelemux) [inline docs/operations/installing_operator_into_cluster_en.md:16]: does this url really change?

Comment (meiserloh) [reply docs/operations/installing_operator_into_cluster_en.md:16]: good catch!

File: controllers/cesregistry/interfaces.go [hunk 3]
```diff
@@ -9,3 +15,34 @@ type doguVersionRegistry interface {
...
+	RegisterNewDogu(ctx context.Context, doguResource *k8sv2.Dogu, dogu *cesappcore.Dogu) error
+	// RegisterDoguVersion registers a new version for an existing dogu in the dogu registry.
+	RegisterDoguVersion(ctx context.Context, dogu *cesappcore.Dogu) error
+	// UnregisterDogu removes a registration of a dogu from the local dogu registry.
+	UnregisterDogu(ctx context.Context, dogu string) error
+}
+
+type K8sClient interface {
+	client.Client
+}
```
[the remark this answers is not in the document]
Comment (meiserloh) [reply controllers/cesregistry/interfaces.go:48]: I reduced the visibility of all interfaces that were not needed outside of its package. Can I find out somehow if they were needed by another repository? We should have a quick look, if I overshot a little and if some interfaces should be public, eventhough they were just used inside the package itself.

File: controllers/dogu_manager.go [hunk 4]
```diff
@@ -32,18 +31,18 @@ var clientSetGetter = func(c *rest.Config) (kubernetes.Interface, error) {
...
+	volumeManager             volumeManager
+	ingressAnnotationsManager additionalIngressAnnotationsManager
+	supportManager            supportManager
+	startStopManager          DoguStartStopManager
 	recorder                  record.EventRecorder
 }
 
 // NewDoguManager creates a new instance of DoguManager
-func NewDoguManager(client client.Client, ecosystemClient ecoSystem.EcoSystemV1Alpha1Interface, operatorConfig *config.OperatorConfig, eventRecorder record.EventRecorder) (*DoguManager, error) {
+func NewDoguManager(client client.Client, ecosystemClient ecoSystem.EcoSystemV2Interface, operatorConfig *config.OperatorConfig, eventRecorder record.EventRecorder) (*DoguManager, error) {
```
Comment (cesmarvin) [inline controllers/dogu_manager.go:45]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'NewDoguManager' is too long (74 > 60)

File: controllers/interfaces.go [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+type execPodFactory interface {
+	exec.ExecPodFactory
+}
+
+type podInterface interface {
+	v1.PodInterface
+}
+
+// premisesChecker includes functionality to check if the premises for an upgrade are met.
+type premisesChecker interface {
```
Comment (cesmarvin) [inline controllers/interfaces.go:300]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `premisesChecker` is unused

File: controllers/interfaces.go [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+	// Check checks if dogu premises are met before a dogu upgrade.
+	Check(ctx context.Context, toDoguResource *v2.Dogu, fromDogu *cesappcore.Dogu, toDogu *cesappcore.Dogu) error
+}
+
+// upgradeExecutor applies upgrades the upgrade from an earlier dogu version to a newer version.
+type upgradeExecutor interface {
```
Comment (cesmarvin) [inline controllers/interfaces.go:306]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `upgradeExecutor` is unused

File: controllers/serviceaccount/interface.go [hunk 7]
```diff
@@ -1,35 +1,75 @@
...
-type SensitiveDoguConfigDeleter interface {
+type sensitiveDoguConfigDeleter interface {
 	DeleteRecursive(ctx context.Context, key string) error
 }
 
-type SensitiveDoguConfig interface {
-	SensitiveDoguConfigGetter
-	SensitiveDoguConfigSetter
-	SensitiveDoguConfigDeleter
+type sensitiveDoguConfig interface {
```
Comment (cesmarvin) [inline controllers/serviceaccount/interface.go:32]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `sensitiveDoguConfig` is unused

File: controllers/serviceaccount/interface.go [hunk 8]
```diff
@@ -1,35 +1,75 @@
...
-type SensitiveDoguConfigProvider interface {
-	GetSensitiveDoguConfig(ctx context.Context, doguName string) (SensitiveDoguConfig, error)
+//nolint:unused
+//goland:noinspection GoUnusedType
+type sensitiveDoguConfigProvider interface {
+	GetSensitiveDoguConfig(ctx context.Context, doguName string) (sensitiveDoguConfig, error)
 }
 
-type SensitiveDoguConfigSetter interface {
+type sensitiveDoguConfigSetter interface {
```
Comment (cesmarvin) [inline controllers/serviceaccount/interface.go:19]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `sensitiveDoguConfigSetter` is unused

File: controllers/serviceaccount/interface.go [hunk 9]
```diff
@@ -1,35 +1,75 @@
...
 	Set(ctx context.Context, key, value string) error
 }
 
-type SensitiveDoguConfigGetter interface {
+type sensitiveDoguConfigGetter interface {
```
Comment (cesmarvin) [inline controllers/serviceaccount/interface.go:23]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `sensitiveDoguConfigGetter` is unused

File: controllers/serviceaccount/interface.go [hunk 10]
```diff
@@ -1,35 +1,75 @@
...
 	Exists(ctx context.Context, key string) (bool, error)
 	Get(ctx context.Context, key string) (string, error)
 }
 
-type SensitiveDoguConfigDeleter interface {
+type sensitiveDoguConfigDeleter interface {
```
Comment (cesmarvin) [inline controllers/serviceaccount/interface.go:28]: 🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type `sensitiveDoguConfigDeleter` is unused

File: docs/operations/installing_operator_into_cluster_de.md [hunk 11]
```diff
@@ -14,7 +14,7 @@ Die Installation von GitHub erfordert die Installations-YAML, die alle benötigt
```
Comment (nhinze23) [inline docs/operations/installing_operator_into_cluster_de.md]: This documentation is very old and should be changed according the helm installation method.

File: docs/operations/installing_operator_into_cluster_en.md [hunk 12]
```diff
@@ -13,7 +13,7 @@ The installation from GitHub requires the installation YAML which contains all r
 
 ```
 GITHUB_VERSION=0.0.6
-kubectl apply -f https://github.com/cloudogu/k8s-dogu-operator/v2/releases/download/v${GITHUB_VERSION}/k8s-dogu-operator_${GITHUB_VERSION}.yaml
```
Comment (nhinze23) [inline docs/operations/installing_operator_into_cluster_en.md]: This documentation is very old and should be changed according the helm installation method.

File: controllers/doguInstallManager.go [hunk 13]
```diff
@@ -69,34 +69,15 @@ func NewDoguInstallManager(client client.Client, mgrSet *util.ManagerSet, eventR
...
```
Comment (cesmarvin) [inline controllers/doguInstallManager.go:72]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Install' has too many statements (48 > 40)

Comment (cesmarvin) [inline controllers/doguInstallManager.go:72]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Install' has too many statements (41 > 40)

File: controllers/doguInstallManager.go (hunk 1 above)
Comment (cesmarvin) [inline controllers/doguInstallManager.go:72]: 🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'Install' has too many statements (59 > 40)

**Pull request merged**